libraries/chain/util/smt_token.cpp

```diff
@@ -138,10 +138,13 @@ flat_map< unit_target_type, share_type > generate_emissions( const smt_token_obj
       else
          new_token_supply = std::max( abs_amount, rel_amount );
 
-      uint32_t unit_sum = emission.emissions_unit.token_unit_sum();
+      if ( token.current_supply + new_token_supply <= token.max_supply )
```

Comment:
This should cap the new tokens to max supply, rather than preventing it if the emissions go above.

For example, a low supply token that constantly replenishes the supply should be able to create the deficit of tokens on emission.